src/AggregateFunctions/AggregateFunctionQuantileExactWeighted.cpp

```diff
@@ -197,23 +232,167 @@ struct QuantileExactWeighted
         }
     }
 
-    /// The same, but in the case of an empty state, NaN is returned.
-    Float64 getFloat(Float64) const
+    /// getFloat implementation without interpolation
+    Float64 getFloatImpl(Float64) const
     {
         throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method getFloat is not implemented for QuantileExact");
     }
 
-    void getManyFloat(const Float64 *, const size_t *, size_t, Float64 *) const
+    /// getManyFloat implementation without interpolation
+    void getManyFloatImpl(const Float64 *, const size_t *, size_t, Float64 *) const
     {
         throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method getManyFloat is not implemented for QuantileExact");
     }
-};
 
+    /// get implementation with interpolation
+    Value getInterpolatedImpl(Float64 level) const
+    {
+        size_t size = map.size();
+        if (0 == size)
+            return std::numeric_limits<Value>::quiet_NaN();
 
-template <typename Value, bool _> using FuncQuantileExactWeighted = AggregateFunctionQuantile<Value, QuantileExactWeighted<Value>, NameQuantileExactWeighted, true, void, false, false>;
-template <typename Value, bool _> using FuncQuantilesExactWeighted = AggregateFunctionQuantile<Value, QuantileExactWeighted<Value>, NameQuantilesExactWeighted, true, void, true, false>;
+        Float64 res = getFloatInterpolatedImpl(level);
+        if constexpr (is_decimal<Value>)
+            return Value(static_cast<typename Value::NativeType>(res));
+        else
+            return static_cast<Value>(res);
+    }
 
-template <template <typename, bool> class Function>
+    /// getMany implementation with interpolation
```

Comment:
It'd make sense to only add this function when using interpolated, otherwise we are compiling things we don't need most of the time. Something like:
```
    template <bool i = interpolated>
    requires(i)
```

Similar in other functions with/without interpolated. The idea is to make it clear to the compiler it can build the expressive functions when we need to, and not always.